### PR TITLE
add domain name functionality

### PR DESCRIPTION
**Pull request by andrewkryshtal**

### CODE REVIEW

File: src/components/account-dropdown/account-dropdown-item/account-dropdown-item.styles.ts [hunk 1]
```diff
@@ -30,5 +30,12 @@ export const useAccountDropdownItemStyles = createUseStyles(({ colors, typograph
   balanceText: {
     ...typography.numbersRegular13,
     color: colors.black
+  },
+  iconContainer: {
+    marginLeft: formatSize(2),
+    paddingHorizontal: formatSize(4),
+    paddingVertical: formatSize(2),
+    backgroundColor: colors.blue10,
+    borderRadius: formatSize(4)
```
Comment (unixvb) [inline src/components/account-dropdown/account-dropdown-item/account-dropdown-item.styles.ts:39]: Seems that these styles are not used

File: src/utils/dns.utils.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { TezosToolkit } from '@taquito/taquito';
+import { isTezosDomainsSupportedNetwork, DomainNameValidationResult, TezosDomainsValidator } from '@tezos-domains/core';
+import { TaquitoTezosDomainsClient } from '@tezos-domains/taquito-client';
+
+export const isTezosDomainsSupported = (tezos: TezosToolkit) =>
+  isTezosDomainsSupportedNetwork('mainnet')
```
Comment (unixvb) [inline src/utils/dns.utils.ts:6]: do we really need this check? we are always using `mainnet`

File: src/components/wallet-address/wallet-address.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      {isString(domainName) ? (
+        <TouchableOpacity
+          style={styles.iconContainer}
+          onPress={e => {
+            e.stopPropagation();
+            setIsShownDomainName(!isShownDomainName);
+          }}>
+          <Icon name={isShownDomainName ? IconNameEnum.Diez : IconNameEnum.Globe} />
+        </TouchableOpacity>
+      ) : null}
```
Comment (unixvb) [inline src/components/wallet-address/wallet-address.tsx:61]: `TouchableIcon`?

File: src/utils/network/rpc-id.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export const NETWORK_IDS = new Map<string, string>([
+  [TempleChainId.Mainnet, 'mainnet'],
+  [TempleChainId.Granadanet, 'granadanet'],
+  [TempleChainId.Florencenet, 'florencenet'],
+  [TempleChainId.Edo2net, 'edo2net'],
+  [TempleChainId.Delphinet, 'delphinet'],
+  [TempleChainId.Carthagenet, 'carthagenet']
+]);
```
Comment (unixvb) [inline src/utils/network/rpc-id.ts:17]: not used

File: src/components/wallet-address/wallet-address.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  const updateDomainReverseName = async (pkh: string) => {
+    const memoizedDomainName = memoize(
+      async () => {
+        return await domainsResolver.resolveAddressToName(pkh);
+      },
+      {
+        cacheKey: () => pkh
+      }
+    );
+    setDomainName((await memoizedDomainName()) ?? '');
```
Comment (unixvb) [inline src/components/wallet-address/wallet-address.tsx:37]: save this with `useState` & `useEffect`

File: src/utils/dns.utils.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { TezosToolkit } from '@taquito/taquito';
+import { DomainNameValidationResult, TezosDomainsValidator } from '@tezos-domains/core';
+import { TaquitoTezosDomainsClient } from '@tezos-domains/taquito-client';
+
+export const isTezosDomainsSupported = (tezos: TezosToolkit) =>
+  new TaquitoTezosDomainsClient({ network: 'mainnet', tezos });
```
Comment (unixvb) [inline src/utils/dns.utils.ts:6]: rename and return resolver directly

File: src/utils/dns.utils.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+
+export const isTezosDomainNameValid = (domain: string) => {
+  return new TezosDomainsValidator().validateDomainName(domain) === DomainNameValidationResult.VALID;
+};
```
Comment (unixvb) [inline src/utils/dns.utils.ts:10]: short

File: src/modals/send-modal/send-modal.tsx [hunk 8]
```diff
@@ -71,12 +77,26 @@ export const SendModal: FC = () => {
...
     ownAccount,
     transferBetweenOwnAccounts
-  }: SendModalFormValues) =>
+  }: SendModalFormValues) => {
+    if (isTezosDomainNameValid(receiverPublicKeyHash)) {
+      setIsLoading(true);
+      const address = await domainsResolver.resolveNameToAddress(receiverPublicKeyHash);
+      if (address !== null) {
+        receiverPublicKeyHash = address;
+        setIsLoading(false);
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:91]: setIsLoading(false); directly after fetching data

**Review by unixvb — APPROVED** [review]

**Review by LendiHop — APPROVED** [review]

**Pull request merged**